ols/src/rag_index/index_loader.py

```diff
@@ -75,26 +90,56 @@ def _set_context(self) -> None:
         Settings.llm = resolve_llm(None)
         logger.info("Setting up storage context for index load...")
         # pylint: disable=W0201
+        if self._vector_store_type == VectorStoreType.FAISS:
+            self._vector_store = FaissVectorStore.from_persist_dir(self._index_path)
+            self._storage_context = StorageContext.from_defaults(
+                vector_store=self._vector_store,
+                persist_dir=self._index_path,
+            )
+        elif self._vector_store_type == VectorStoreType.POSTGRES:
+            postgres = self._index_config.postgres
+            user = postgres.user
+            password = postgres.password
+            host = postgres.host
+            port = postgres.port
+            dbname = postgres.dbname
+            table_name = self._index_id.replace("-", "_")
+            embed_dim = len(Settings.embed_model.get_text_embedding("random text"))
```

Comment:
This should be stored once we know what model we are using. Not derived during runtime. 
Maybe https://docs.llamaindex.ai/en/stable/api_reference/embeddings/huggingface/#llama_index.embeddings.huggingface.HuggingFaceInferenceAPIEmbedding.get_model_info would do the trick?

Reply:
The function that you suggested seems to retrieve metadata from Hugging Face ([link](https://github.com/run-llama/llama_index/blob/main/llama-index-integrations/embeddings/llama-index-embeddings-huggingface-api/llama_index/embeddings/huggingface_api/base.py#L137-L139)), which is not what we want to do.

`Settings.embed_model` contains an instance of `HuggingFaceEmbedding`. While I did check it on debugger, I could not see fields that clearly shows its embeddings dimension, which is `768`.
![image](https://github.com/user-attachments/assets/c1fee813-0516-4e65-9d83-d34357ef41b7)

Since this is called only at the initialization of the vectore store, I think keeping this code won't hurt runtime performance alot. What do you think?

Reply:
My bad HFapi/HF. But looking at the code, the `_model` field in `HuggingFaceEmbedding` is a `SentenceTransformer` and that object has [get_sentence_embedding_dimension](https://www.sbert.net/docs/package_reference/sentence_transformer/SentenceTransformer.html#sentence_transformers.SentenceTransformer.get_sentence_embedding_dimension). Thar should work.

If it doesn't, I would still rather prefer that the embedding dimension was in the config, rather than in the code. Embedding dimension is a very important characteristic and it should be obvious at first glance.

Reply:
Updated the code to use s get_sentence_embedding_dimension.  Thanks.